controllers/routemonitor/routemonitor_supplement.go

```diff
@@ -4,40 +4,32 @@ import (
 	"context"
 
 	"github.com/openshift/route-monitor-operator/api/v1alpha1"
+	"github.com/openshift/route-monitor-operator/pkg/const/blackbox"
 	utilreconcile "github.com/openshift/route-monitor-operator/pkg/util/reconcile"
 )
 
-func (r *RouteMonitorReconciler) EnsureBlackBoxExporterResourcesExists(ctx context.Context) error {
-	if err := r.EnsureBlackBoxExporterDeploymentExists(ctx); err != nil {
-		return err
-	}
-	// Creating Service after because:
-	//
-	// A Service should not point to an empty target (Deployment)
-	if err := r.EnsureBlackBoxExporterServiceExists(ctx); err != nil {
-		return err
-	}
-	return nil
+//go:generate mockgen -source $GOFILE -destination ../../pkg/util/test/generated/mocks/$GOPACKAGE/blackboxexporter.go -package $GOPACKAGE BlackboxExporter
+type BlackboxExporter interface {
+	EnsureBlackBoxExporterResourcesExist() error
+	EnsureBlackBoxExporterResourcesAbsent() error
+	ShouldDeleteBlackBoxExporterResources() (blackbox.ShouldDeleteBlackBoxExporter, error)
 }
 
 func (r *RouteMonitorReconciler) EnsureRouteMonitorAndDependenciesAbsent(ctx context.Context, routeMonitor v1alpha1.RouteMonitor) (utilreconcile.Result, error) {
 	log := r.Log.WithName("Delete")
 
-	shouldDeleteBlackBoxResources, err := r.ShouldDeleteBlackBoxExporterResources(ctx, routeMonitor)
+	shouldDeleteBlackBoxResources, err := r.BlackboxExporter.ShouldDeleteBlackBoxExporterResources()
```

Comment:
it's used here but not initialized in the Reconcile, might cause issues

Reply:
all dependencies of `RouteMonitorReconcile` are injected from `main.go`, I'd prefer to do that in the Reconcile as well, but I didn't want to change the whole existing code structure.

Reply:
well, that can be a next step refactor.

also I think I just missed the main.go injection 😅